src/iterative_sorting/iterative_sorting.py

```diff
@@ -6,24 +6,28 @@ def selection_sort( arr ):
         smallest_index = cur_index
         # TO-DO: find next smallest element
         # (hint, can do in 3 loc) 
-             
-
-
+        for j in range(i, len(arr)):
+            if arr[j] < arr[smallest_index]:
+                smallest_index = j
 
         # TO-DO: swap
-
-
-
+        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]
 
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort( arr ):
 
+    swap = True
+    while swap:
+        swap = False
+        for i in range(len(arr) - 1):
+            if arr[i] > arr[i+1]:
+                arr[i], arr[i+1] = arr[i+1], arr[i]
+                swap = True
     return arr
 
```

Comment:
Yes!  Its great to see that you went with the boolean/while approach for this.  This is much more optimal as it will end the loop as soon as it sees the list is sorted.